model_compression_toolkit/quantizers_infrastructure/README.md

```diff
@@ -0,0 +1,93 @@
+# Quantizers Infrastructure (QI)
+
+Quantizers infrastructure is a library that provides quantization modules for hardware-oriented model optimization tools.
+
+The quantization modules (quantizers) can be used for emulating inference-time quantization (inferable-quantizers) and for optimization of model quantization during training (trainable quantizers).
+
+## Library structure
+
+### Quantizer
+
+Quantizer instance implements the quantization function: conversion of floating point values to quantized values represented in floating point (fake-quantization).
+Quantizer instance should be defined per layer, separately for weights quantization and activations quantization.
+
+The quantizers are divided into two main types:
+
+`BaseInferableQuantizer` - A type of quantizer that uses for emulating inference-time quantization   
+
+`BaseTrainableQuantizer` - A type of quantizer that uses for quantization optimization during training. This type of quantizer contains learnable parameters
+
+Usage example:
+
+```python
+from model_compression_toolkit import quantizers_infrastructure as qi
+
+# generate a signed 8-bits symmetric quantizer for numbers in the interval [-2,2) 
+w_quantizer = qi.WeightsSymmetricInferableQuantizer(num_bits=8,
+                                                    threshold=2,
+                                                    signed=True,
+                                                    per_channel=True)
+```
+### Dispatcher
+
+`NodeQuantizationDispatcher` instance contains a list of relevant quantizers for node (layer) in the model:
+weights quantizer per attribute (kernel, bias, ...)
+activation quantizer per layer's output
+
+Usage example:
+
+```python
+# generate node dispatcher with the above weights quantizer 
+dispatcher = qi.KerasNodeQuantizationDispatcher(w_quantizer)
+```
+### Quantization Wrapper
+
+A quantization wrapper instance is used as a layer's replacement which emulates the layer's operation only with quantization.
+The quantization wrapper receives the layer to wrap, and a dispatcher for mapping the relevant quantizer per attribute. 
+
+`KerasQuantizationWrapper` - used to wrap Keras layer
+
+`PytorchQuantizationWrapper` - used to wrap torch nn module
+
+Usage example:
+
+```python
+from tensorflow import keras
+
+# generate an example of Keras model
+inputs = keras.layers.Input(shape=(32,32,3))
+x = keras.layers.Conv2D(6, 7, use_bias=False)(inputs)
+model = keras.Model(inputs=inputs, outputs=x)
+
+# create quantization wrapper from the convolution layer according to the dispatcher
+qi.KerasQuantizationWrapper(layer=model.layers[1], dispatcher)
```

Comment:
Don't we want the "clone" code for actually replacing the layer with the wrapped layer?